File: basicprogramming2/basicprogramming2.py
def main():
    n, t = list(map(int, input().split()))
    items = list(map(int, input().split()))

    if t == 1:
        items.sort()
        for item in items:
            c = 7777-item
            if c in items:
                print("Yes")
                return
            # for j in range(i + 1, n):
            #     if items[i] + items[j] == 7777:
            #         print("Yes")
            #         return
        # for i in range(n):
        #     for j in range(i + 1, n):
        #         if items[i] + items[j] == 7777:
        #             print("Yes")
        #             return
        print("No")
    elif t == 2:
        if len(set(items)) == len(items): print("Unique")
        else: print("Contains duplicate")
    elif t == 3:
        # if len(set(items)) < len(items) // 2: print(-1)
        # else: print("Contains duplicate")
        counts = {}
        for item in items:
            if item in counts:
                counts[item]+=1
            else:
                counts[item]=1
        for k, v in counts.items():
            if v > len(items) // 2:
                print(k)
                break
            else:
                print(-1)

    elif t == 4:
        items.sort()
        if len(items) == 1:
            print(items[0])
        elif len(items) % 2:
            print(items[len(items)//2 - 1])
        else:
            print(items[len(items)//2 - 1], items[len(items)//2])
    elif t == 5:
        newList = []
        for item in items:
            if 100 <= item <= 999:
                newList.append(item)

        newList.sort()
        print(' '.join(map(str, newList)))
    return

if __name__ == "__main__":
    main()

File: basicprogramming2/basicprogramming2.cpp
#include <bits/stdc++.h>

typedef long long ll;
using namespace std;

int main() {
    int n, t;
    cin >> n >> t;
    vector<int> array;
    for (int i = 0; i < n; i++) {
        int tmp;
        cin >> tmp;
        array.push_back(tmp);
    }

    switch (t) {
        case 1: {
            sort(array.begin(), array.end());

            for ( auto item : array ) {
                if (binary_search(array.begin(), array.end(), 7777-item)) {
                    cout << "Yes\n";
                    return 0;
                }
            }
            cout << "No\n";
            break;
        }
        case 2: {
            unordered_set<int> map;
            for (int i : array) {
                if (map.find(i) == map.end()) // Not found
                    map.insert(i);
                else {
                    cout << "Contains duplicate\n";
                    return 0;
                }
            }
            cout << "Unique\n";
            break;
        }
        case 3: {
            unordered_map<int, int> map;
            int size = array.size();
            for (int i : array) {
                if (map.find(i) == map.end()) { // Not found
                    map.insert({i, 1});
                } else {
                    map[i]++;
                    if (map[i] > size / 2) {
                        cout << i << endl;
                        return 0;
                    }
                }
            }
            cout << "-1\n";
            break;
        }
        case 4: {
            sort(array.begin(), array.end());
            int size = array.size();
            if (size & 1) { // Odd
                cout << array[size / 2] << endl;
            } else {
                cout << array[size / 2 - 1] << ' ' << array[size / 2] << '\n';
            }
            break;
        }
        case 5: {
            sort(array.begin(), array.end());
            for (int i : array) {
                if (100 <= i && i <= 999)
                    cout << i << " ";
            }
            break;
        }
        default:
            break;
    }
    
    return 0;
}
